lib/elixir/lib/kernel/special_forms.ex

```diff
@@ -205,99 +210,59 @@ defmodule Kernel.SpecialForms do
-  latter meaning it will be resolved at VM load time).
+  ## Options
 
-  Many options can be given by using `-` as separator, order is
-  arbitrary. The following are all the same:
+  Many options can be given by using `-` as separator. Order is
+  arbitrary, so the following are all equivalent:
 
       <<102 :: integer-native, rest :: binary>>
-      <<102 :: native-integer, rest :: binary>>  
+      <<102 :: native-integer, rest :: binary>>
       <<102 :: unsigned-big-integer, rest :: binary>>
       <<102 :: unsigned-big-integer-size(8), rest :: binary>>
       <<102 :: unsigned-big-integer-8, rest :: binary>>
       <<102 :: 8-integer-big-unsigned, rest :: binary>>
       <<102, rest :: binary>>
 
-  And so on.
-
-  Endianness only makes sense for integers and some UTF code
-  point types (utf16 and utf32).
-
-  Finally, we can also specify size and unit for each segment. The
-  unit is multiplied by the size to give the effective size of
-  the segment in bits. The default unit for integers, floats,
-  and bitstrings is 1. For binaries, it is 8.
+  ### Unit and Size
 
-  Since integers are default, the default unit is 1. The example below
-  matches because the string "foo" takes 24 bits and we match it
-  against a segment of 24 bits, 8 of which are taken by the integer
-  102 and the remaining 16 bits are specified on the rest.
+  The length of the match is equal to the `unit` (a number of bits) times the
+  `size` (the number of repeated segnments of length `unit`).
 
-      iex> <<102, _rest :: size(16)>> = "foo"
-      "foo"
-
-  We can also match by specifying size and unit explicitly:
+  Type      | Default Unit
+  --------- | ------------
+  `integer` | 1 bit
+  `float`   | 1 bit
+  `binary`  | 8 bits
 
-      iex> <<102, _rest :: size(2)-unit(8)>> = "foo"
-      "foo"
+  Sizes for types are a bit more nuanced. The default size for integers is 8.
 
-  However, if we expect a size of 32, it won't match:
+  For floats, it is 64. For floats, `size * unit` must result in 32 or 64,
+  corresponding to [IEEE 754](http://en.wikipedia.org/wiki/IEEE_floating_point)
+  binary32 and binary64, respectively.
 
-      iex> <<102, _rest :: size(32)>> = "foo"
-      ** (MatchError) no match of right hand side value: "foo"
+  For binaries, the default is the size of the binary. Only the last binary in a
+  binary match can use the default size. All others must have their size
+  specified explicitly, even if the match is unambiguous.
 
-  Size and unit are not applicable to utf8, utf16, and utf32.
+  For example:
 
-  The default size for integers is 8. For floats, it is 64. For
-  binaries, it is the size of the binary. Only the last binary
-  in a binary match can use the default size (all others must
-  have their size specified explicitly).
+      iex> <<name::binary, " the ", species::binary>>= <<"Frank the Walrus">>
+      ** (CompileError): a binary field without size is only allowed at the end of a binary pattern
```

Comment:
This failure comes out of the doctest boundaries and makes whole test run to fail.